flake.nix

```diff
@@ -0,0 +1,74 @@
+{
+  description = "Olean binary format parser in Lean";
+
+  inputs = {
+    lean = {
+      url = github:leanprover/lean4;
```

Comment:
Can you pin this to the same nightly as the `lean-toolchain` file?

Reply:
I couldn't set it specifically since nix uses git and there are no nightly tags. So I sat it to the latest commit.